### ISSUE TITLE
Is there a way to exclude the cache tree digest from the key when caching?

### ISSUE THREAD

**Issue by tez123z** [rails/jbuilder#411]
Is there a way to exclude the cache tree digest from the key when caching?

jbuilder/v3/users/38-20170809004706672937000/performers/1493168-20170823184135817877000/d84abdae117cd787b8204fd744e7dc0

In the case above it would be the last chunk "d84abdae117cd787b8204fd744e7dc0"

I'm trying to implement a cache structure with jbuilder where every object is cached and the cache is wiped after every update. But I need a cache key consistent with the model cache keys to find and expire the key. (Rails.cache.delete_matched isn't working - nor are any View caching helpers).

So i've come to the conclusion that the I need the digest to be remove so i can regenerate the exact cache key for easy deleting.

Any help would be very much appreciated!

**Comment by codabrink** [comment]
The option you're looking for is `skip_digest: true`

```ruby
json.cache! "my_key_#{id}", skip_digest: true do
   ...
end
```

**Comment by robin850** [comment]
Yes, @codabrink is right ; under the hood [`cache_fragment_name`](https://api.rubyonrails.org/classes/ActionView/Helpers/CacheHelper.html#method-i-cache_fragment_name) is used so you can use `skip_digest` or `virtual_path` if you want.
